### PR TITLE
subsys: Bluetooth: LLL: Add functions that allow configure Radio for reception and sampling of CTE

### PR DESCRIPTION
**Pull request by ppryga-nordic**
The PR includes:
- fix for build error from direction finding compilation when `IS_ENABLE(CONFIG_BT_CTLR_DF_ADV_CTE_TX)`  is `false`
- API to configure Radio to receive and sample CTE

Part of former code responsible for configuration of CTE Tx was refactored to comply with code responsible for CTE Rx configuration.

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_df.c [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #define LOG_MODULE_NAME bt_ctlr_ull_df
 #include "common/log.h"
 
+#if IS_ENABLED(CONFIG_BT_CTLR_DF_ADV_CTE_TX)
```
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_df.c:34]: For consistency use '#if defined' instead of '#if IS_ENABLED'
See also PR #32609

**Review by kruithofa — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_df.c (hunk 1 above)
Comment (ppryga) [reply subsys/bluetooth/controller/ll_sw/ull_df.c]: Corrected. I'll create a separate PR to clean this in all DF related files (if IS_ENABLES is still available)

**Comment by ppryga** [comment]
Added change requested by @kruithofa.
Rebased onto master.

**Comment by ppryga** [comment]
Removed cte_length argument from functions: `radio_df_cte_rx_2us_switching`, `radio_df_cte_rx_4us_switching`.
`cte_length` is not used when CTEINLINE is enabled. In such situation CTE length is send in `CTEInfo` filed in PDU.

**Comment by ppryga** [comment]
Fixed compliance issue

File: subsys/bluetooth/controller/ll_sw/ull_df.c [hunk 2]
```diff
@@ -46,6 +47,7 @@
 
 static struct lll_df_adv_cfg lll_df_adv_cfg_pool[CONFIG_BT_CTLR_ADV_AUX_SET];
 static void *df_adv_cfg_free;
+#endif /* #if CONFIG_BT_CTLR_DF_ADV_CTE_TX */
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_df.c:50]:
```suggestion
#endif /* CONFIG_BT_CTLR_DF_ADV_CTE_TX */
```

File: subsys/bluetooth/controller/ll_sw/ull_df.c [hunk 3]
```diff
@@ -54,13 +56,15 @@ static void *df_adv_cfg_free;
...
 
+#if defined(CONFIG_BT_CTLR_DF_ADV_CTE_TX)
 /* @brief Function acquires memory for DF advertising configuration.
  *
  * The memory is acquired from private @ref lll_df_adv_cfg_pool memory store.
  *
  * @return Pointer to lll_df_adv_cfg or NULL if there is no more free memory.
  */
 static struct lll_df_adv_cfg *df_adv_cfg_acquire(void);
+#endif /* #if CONFIG_BT_CTLR_DF_ADV_CTE_TX */
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_df.c:67]:
```suggestion
#endif /* CONFIG_BT_CTLR_DF_ADV_CTE_TX */
```

File: subsys/bluetooth/controller/ll_sw/ull_df.c [hunk 4]
```diff
@@ -96,11 +100,12 @@ int ull_df_reset(void)
 
 static int init_reset(void)
 {
+#if defined(CONFIG_BT_CTLR_DF_ADV_CTE_TX)
 	/* Initialize advertising DF memory configuration pool. */
 	mem_init(lll_df_adv_cfg_pool, sizeof(struct lll_df_adv_cfg),
 		 sizeof(lll_df_adv_cfg_pool) / sizeof(struct lll_df_adv_cfg),
 		 &df_adv_cfg_free);
-
+#endif /* #if CONFIG_BT_CTLR_DF_ADV_CTE_TX */
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_df.c:108]:
```suggestion
#endif /* CONFIG_BT_CTLR_DF_ADV_CTE_TX */
```

**Review by cvinayak — CHANGES_REQUESTED** [review]
nitpicks, otherwise looks good.

File: subsys/bluetooth/controller/ll_sw/ull_df.c (hunk 2 above)
Comment (ppryga) [reply subsys/bluetooth/controller/ll_sw/ull_df.c]: Corrected

File: subsys/bluetooth/controller/ll_sw/ull_df.c (hunk 3 above)
Comment (ppryga) [reply subsys/bluetooth/controller/ll_sw/ull_df.c]: Corrected

File: subsys/bluetooth/controller/ll_sw/ull_df.c (hunk 4 above)
Comment (ppryga) [reply subsys/bluetooth/controller/ll_sw/ull_df.c]: Corrected

**Comment by ppryga** [comment]
Added one commit with:
- removed undefined symbols error
- added missing symbols to radio stubs implementation.

@cvinayak your requested modifications are implemented.

**Comment by carlescufi** [comment]
@cvinayak @thoh-ot @kruithofa please take another look

**Review by cvinayak — APPROVED** [review]
LGTM.

**Review by kruithofa — APPROVED** [review]
LGTM

**Pull request merged**